gobblin-compaction/src/main/java/gobblin/compaction/mapreduce/MRCompactor.java

```diff
@@ -46,16 +50,46 @@
 import gobblin.util.DatasetFilterUtils;
 import gobblin.util.HadoopUtils;
 
+import static gobblin.compaction.mapreduce.MRCompactor.FolderState.*;
+import static gobblin.compaction.mapreduce.MRCompactorJobRunner.Status.*;
+
 
 /**
  * MapReduce-based {@link gobblin.compaction.Compactor}.
  *
  * @author ziliu
  */
+
 public class MRCompactor implements Compactor {
 
   private static final Logger LOG = LoggerFactory.getLogger(MRCompactor.class);
-  private static final List<Job> RUNNING_MR_JOBS = Lists.newCopyOnWriteArrayList();
+
+  private static final double HIGH_PRIORITY = 3.0;
+  private static final double NORMAL_PRIORITY = 2.0;
+  private static final double LOW_PRIORITY = 1.0;
+  private static final double PRIORITY_REDUCTION_FACTOR = 1.0 / 3.0;
+  private static final long COMPACTION_JOB_WAIT_INTERVAL_SECONDS = 10;
+  private static final Map<String, Job> RUNNING_MR_JOBS = Maps.newConcurrentMap();
```

Comment:
Then I'll need to pass an instance of MRCompactor to `MRCompactorJobRunner` which creates the Hadoop job.